### ISSUE TITLE
Dedicated building types

### ISSUE THREAD

**Issue by Matthias84** [StrandedKitty/streets-gl#40]
Hi,

I would ask if its possible to add nice specific defaults for buildings that only have a building type, but nothing set for roof and levels or material?

I suggest:

- garage
- garages
- barn / farm_auxiliary / stable
- bungalow / cabin
- detached / semiattached
- greenhouse
- residential
- riding_hall 
- sports_hall
- stadium 
- hangar 
- parking 
- service 
- roof (esp. for fuel stations)

**Comment by hb0nd** [comment]
Honestly, I think anything more than changing the style of windows etc shouldn't be inferred. having things appear as blank boxes incentivises mappers to add the correct tagging

**Comment by Matthias84** [comment]
Yes but the discussion of implicite vs. explicite is pretty old on OSM side and both side have pretty good arguments for it's position.

My idea was esp. to remove windows for service like buildings (garages, ...) to increase the plausibility. This is how osm2world does it since years. Maybe some default materials would be nice, but  I agree that its hard to find global common appearance esp. between continents ...

**Comment by artus70** [comment]
Yeah, windows should be drawn only on appropriate building types!
Or a tag was needed to describe whether a building or building:part has windows or not.

**Comment by ivanbranco** [comment]
> Or a tag was needed to describe whether a building or building:part has windows or not.

window=no should be already supported

**Comment by artus70** [comment]
> window=no should be already supported

This is great, thanks! Is this documented anywhere?

**Comment by HolgerJeromin** [comment]
>> window=no should be already supported

> This is great, thanks! Is this documented anywhere?

https://wiki.openstreetmap.org/wiki/Key:window

**Comment by hb0nd** [comment]
Windows are already excluded from certain building types (but is overridden with explicit tagging): 

https://github.com/StrandedKitty/streets-gl/blob/cd53e4ef7c93cb80e08344ae12d6163bcb16229c/src/lib/tile-processing/vector/qualifiers/factories/helpers/isBuildingHasWindows.ts#L1-L20

**Comment by artus70** [comment]
@HolgerJeromin and @endim8: Very cool, thanks for the hints!

@endim8 One could discuss whether `building=yes` or `building:part=yes` should be drawn w/o windows, too.

**Comment by Matthias84** [comment]
Any chance how e.g. greenhouse could become a default glass material? :thinking:
